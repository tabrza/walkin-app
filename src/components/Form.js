import React, { Component } from 'react';

class Form extends Component {
  // constructor(props) {
  //   super(props);
  // }
  state = {
    person: {
      name: '',
      waitTime: 0,
      origTime: 0
    }
  }

  // set the content to the value in the input box
  userInputHandler = (e) => {
    const newPerson = { ...this.state.person };
    newPerson[e.target.name] = e.target.value;
    if (e.target.name === 'waitTime') newPerson.origTime = e.target.value;
    this.setState({
      person: newPerson
    });
  };

  addPersonHandler = () => {
    this.props.addPerson(this.state.person);
    const resetPerson = { ...this.state.person };
    resetPerson.name = '';
    resetPerson.waitTime = 0;
    resetPerson.origTime = 0;
    // for (const prop of resetPerson.getOwnPropertyNames(obj)) {
    //   delete obj[prop];
    // }
    this.setState({
      person: resetPerson
    });
  };

  render() {
    return (
      <div className="input-group form">
        <input
          name="name"
          className="input-group__item text-input"
          placeholder="Name"
          value={this.state.person.name}
          onChange={this.userInputHandler}
        />
        <input
          name="waitTime"
          className="input-group__item text-input"
          placeholder="Wait Time"
          value={this.state.person.time}
          onChange={this.userInputHandler}
        />
        <button className="button" onClick={this.addPersonHandler}>Save</button>
      </div>
    );
  }
}


export default Form;
