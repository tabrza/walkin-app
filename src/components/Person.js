import React from 'react';
import ProgressBar from './ProgressBar';

const person = props => (
  <div className="list-header">
    <div className="list-body">
      <div className="list-item">
        <p className=""><strong> { props.name } </strong></p>
        <p className=""><strong> { props.time } </strong></p>
        <div className="list-item__data">
          <ProgressBar time={props.time} />
        </div>
      </div>
    </div>
  </div>
);

export default person;
