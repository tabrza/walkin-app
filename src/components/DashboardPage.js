import React, { Component } from 'react';
import Person from './Person';
import Form from './Form';
import { firebase, database } from '../firebase/firebase';

class DashboardPage extends Component {
  constructor() {
    super();
    this.db = database.ref().child('persons');
  }
  state = {
    persons: []
  }

  componentWillMount() {
    const updatedList = [...this.state.persons];
    // DB snapshot
    this.db.on('child_added', (snap) => {
      updatedList.push({
        id: snap.key,
        name: snap.val().person.name,
        waitTime: snap.val().person.waitTime,
        origTime: snap.val().person.origTime
      });

      this.setState({
        persons: updatedList
      });
    });
  }

  componentDidMount() {
    this.interval = setInterval(() => this.updateTime(), 60000);
  }

  updateTime = () => {
    const list = [...this.state.persons];
    const updatedList = [];

    for (let i = 0; i < list.length; i += 1) {
      const person = list[i];
      let newTime = parseInt(person.waitTime, 10);
      // console.log('person in array',person);
      if (newTime > 0) {
        newTime -= 1;
      }

      const object = {
        id: person.id,
        name: person.name,
        waitTime: newTime,
        origTime: person.origTime
      };

      updatedList.push(object);

      // Write`/persons/${  person.id  }/person/`n the posts list and the user's post list.
      const updates = {};
      updates[`/persons/${person.id}/person/`] = object;

      firebase.database().ref().update(updates);
    }

    this.setState({
      persons: updatedList
    });
  }

  addPerson = (person) => {
    // console.log('person',person);
    this.db.push().set({
      person
    });
  };

  render() {
    // console.log('state',this.state.persons);
    const allPersons = this.state.persons.map(person => (
      // subtract person wait time with counter
      <Person
        key={person.id}
        name={person.name}
        waitTime={person.waitTime}
        origTime={person.origTime}
      />
    ));

    return (
      <div className="content-container">
        <Form addPerson={this.addPerson} />
        {allPersons}
      </div>
    );
  }
}

export default DashboardPage;
