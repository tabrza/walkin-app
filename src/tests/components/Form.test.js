import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import DashboardPage from '../../components/DashboardPage';
import Person from '../../components/Person';
import Form from '../../components/Form';

configure({ adapter: new Adapter() }); // connect enzyme


describe('<Form />', () => {
  it('renders without crashing', () => {
    shallow(<Form />);
  });

  it('should render Form correctly', () => {
    const wrapper = shallow(<Form />);
    expect(wrapper).toMatchSnapshot();
  });
});
