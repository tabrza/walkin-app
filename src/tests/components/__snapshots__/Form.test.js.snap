// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should render Form correctly 1`] = `
<div
  className="input-group form"
>
  <input
    className="input-group__item text-input"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    value=""
  />
  <input
    className="input-group__item text-input"
    name="waitTime"
    onChange={[Function]}
    placeholder="Wait Time"
  />
  <button
    className="button"
    onClick={[Function]}
  >
    Save
  </button>
</div>
`;
